# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLCanvasElement.h
# Commit: 047bcbf3769f
# Full Hash: 047bcbf3769f8dcdeda5f00e8dd1b0cad893e602
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2021-12-10 21:58:52
# Regressor Bug: 1736177
# File Overlap Count: 2
# Description:
#   Bug 1736177 - Part 8. Remove unused visibility observer from HTMLCanvasElement. r=jgilbert
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D130784
# ==============================================================================

diff -r 2859a91f3a73 -r 047bcbf3769f dom/html/HTMLCanvasElement.h
--- a/dom/html/HTMLCanvasElement.h	Fri Dec 10 02:57:54 2021 +0000
+++ b/dom/html/HTMLCanvasElement.h	Fri Dec 10 02:57:54 2021 +0000
@@ -58,19 +58,14 @@
 
 // Listen visibilitychange and memory-pressure event and inform
 // context when event is fired.
-class HTMLCanvasElementObserver final : public nsIObserver,
-                                        public nsIDOMEventListener {
+class HTMLCanvasElementObserver final : public nsIObserver {
  public:
   NS_DECL_ISUPPORTS
   NS_DECL_NSIOBSERVER
-  NS_DECL_NSIDOMEVENTLISTENER
 
   explicit HTMLCanvasElementObserver(HTMLCanvasElement* aElement);
   void Destroy();
 
-  void RegisterVisibilityChangeEvent();
-  void UnregisterVisibilityChangeEvent();
-
   void RegisterObserverEvents();
   void UnregisterObserverEvents();
 
@@ -319,7 +314,6 @@
 
   layers::LayersBackend GetCompositorBackendType() const;
 
-  void OnVisibilityChange();
   void OnMemoryPressure();
   void OnDeviceReset();
 
